src/main/scala/lms/core/stub.scala

```diff
@@ -825,6 +840,16 @@ trait PrimitiveOps extends Base with OverloadHack {
     def toLong(implicit __pos: SourceContext): Rep[Long] = cast_helper[Char,Long](self)
   }
 
+  // Generic?
+  implicit class GenericNegate[T:Numeric:Manifest](val self: Rep[T])(implicit __pos: SourceContext) {
+    def unary_- = Arithmetic.unary_-[T](self)
+
+    def +(rhs: Rep[T])(implicit __pos: SourceContext, __imp1: Overloaded1) = { Arithmetic.plus[T](self, rhs) }
+    def -(rhs: Rep[T])(implicit __pos: SourceContext, __imp1: Overloaded1) = { Arithmetic.minus[T](self, rhs) }
+    def *(rhs: Rep[T])(implicit __pos: SourceContext, __imp1: Overloaded1) = { Arithmetic.times[T](self, rhs) }
+    def /(rhs: Rep[T])(implicit __pos: SourceContext, __imp1: Overloaded1) = { Arithmetic.divide[T](self, rhs) }
+  }
```

Comment:
unfortunately this has duplications with what already in the code base

Reply:
Well, the even more unfortunate thing is that this duplication works in my terminal (maybe due to my higher java and sbt version) but doesn't pass gate test in here. 

How do we get this Numeric type working for Primitives such as + - * / ?